internal/metrics/synthetics/token.go

```diff
@@ -0,0 +1,140 @@
+package synthetics
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/superfly/fly-go"
+	"github.com/superfly/flyctl/gql"
+	"github.com/superfly/flyctl/internal/config"
+	"github.com/superfly/flyctl/internal/flyutil"
+	"github.com/superfly/flyctl/internal/state"
+	"github.com/superfly/flyctl/terminal"
+	"github.com/superfly/macaroon"
+	"github.com/superfly/macaroon/flyio"
+	"github.com/superfly/macaroon/resset"
+)
+
+func generateSyntheticsToken(ctx context.Context) (token string, err error) {
+	client := flyutil.ClientFromContext(ctx)
+	var orgs []fly.Organization
+	if orgs, err = client.GetOrganizations(ctx); err != nil {
+		return
+	}
+
+	var orgsTokens [][]byte
+
+	for _, org := range orgs {
+		var orgToken [][]byte
+
+		orgToken, err = getOrgReadOnlyToken(ctx, org)
+		if err != nil {
+			return
+		}
+
+		orgsTokens = append(orgsTokens, orgToken...)
+	}
+
+	authHeader := macaroon.ToAuthorizationHeader(orgsTokens...)
+
+	return authHeader, nil
+}
+
+func GetSyntheticsToken(ctx context.Context) (token string, err error) {
+	// Prevent synthetics panics from bubbling up to the user.
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("panic: %v", r)
+		}
+	}()
+
+	cfg := config.FromContext(ctx)
+	if cfg.SyntheticsToken != "" {
+		terminal.Debugf("Config has synthetics token\n")
+		return cfg.SyntheticsToken, nil
```

Comment:
As the user is added to and removed from orgs, this set of tokens wont change. Is that okay?

Reply:
Good point!

For removing users from Orgs, if the token is revoked it will fail to read the organization, correct?

For adding a user to an org, maybe I can add a command to refresh the token or refresh it if the count of monitored orgs change. Do you have a recommendation?

Also, if we are adding a new token profile, maybe I can lower the expire to something like 1h or 3h and let the token to be refreshed on reconnect?

Reply:
Having them expire quickly and not persisting them in the config file definitely sounds nicer.

Reply:
Reduced time to expire to 1h and persist only in memory to avoid having to generate a new one on reconnects which happen frequently.